kornia/augmentation/presets/ada.py

```diff
@@ -0,0 +1,263 @@
+
+    adjust a global probability p over all augmentations list to select a subset of images to augment
+    based on an exponential moving average of the Discriminator's accuracy labeling real samples.
+
+    Args:
+        *args: a list of kornia augmentation modules, set to a default list if not specified.
+
+        initial_p: initial global probability `p` for applying the augmentations on
+
+        adjustment_speed: step size for updating the global probability `p`
+
+        max_p: the maximum value to clamp `p` at
+
+        target_real_acc: target Discriminator accuracy to prevent overfitting
+
+        ema_lambda: EMA smoothing factor to compute the $
+        \mathrm{ema_real_accuracy} = \lambda_\text{EMA} * mathrm{real_accuracy} +
+        (1 - \lambda_\text{EMA}) * mathrm{real_accuracy}
+        $
+
+        update_every: `p` update frequency
+
+        crop_size: the used in the `RandomCrop` default augmentation
+
+        same_on_batch: apply the same transformation across the batch.
+        If None, it will not overwrite the function-wise settings.
+
+        data_keys: the input type sequential for applying augmentations. Accepts "input", "image", "mask",
+            "bbox", "bbox_xyxy", "bbox_xywh", "keypoints", "class", "label".
+
+        **kwargs: the rest of the `kwargs` passed to the `AugmentationSequential` attribute containing augmentation
+
+    Examples:
+        >>> from kornia.augmentation.presets.ada import AdaptiveDiscriminatorAugmentation
+        >>> original = torch.randn(2, 3, 5, 6)
+        >>> ada = AdaptiveDiscriminatorAugmentation()
+        >>> augmented = ada(original)
+
+    This example demonstrates using default augmentations with AdaptiveDiscriminatorAugmentation in a GAN training loop.
+
+
+        >>> import kornia.augmentation as K
+        >>> from kornia.augmentation.presets.ada import AdaptiveDiscriminatorAugmentation
+        >>> originals = torch.randn(2, 3, 5, 6)
+        >>> aug_list = [
+        ...     K.RandomRotation90(times=(0, 3), p=1),
+        ...     K.RandomAffine(degrees=10, translate=(.1, .1), scale=(.9, 1.1), p=1),
+        ...     K.ColorJitter(brightness=.2, contrast=.2, saturation=.2, hue=.1, p=1),
+        ... ]
+
+        >>> ada = AdaptiveDiscriminatorAugmentation(*aug_list)
+        >>> augmented = ada(original)
+
+    This example demonstrates using custom augmentations with AdaptiveDiscriminatorAugmentation.
+    """
+
+    def __init__(
+        self,
+        *args: Union[_AugmentationBase, ImageSequential],
+        initial_p: float = 1e-5,
+        adjustment_speed: float = 1e-2,
+        max_p: float = 0.8,
+        target_real_acc: float = 0.85,
+        ema_lambda: float = 0.99,
+        update_every: int = 5,
+        crop_size: Tuple[int, int] = (64, 64),
+        data_keys: Optional[_data_keys_type] = None,
+        same_on_batch: Optional[bool] = False,
+        **kwargs: Any,
+    ) -> None:
+        if not args:
+            args = (
```

Comment:
Create a new variable, this updates the inputs which might introduce bugs to the user. I would also define outside a "default" function returning this sequential